[FUNK-1550] IP Allowlisting Changes (#7244)
segmentio/segment-docs: src/connections/functions/index.md

With Functions Copilot, you can instrument custom integrations, enrich and transform data, and even secure sensitive data nearly instantaneously without writing a line of code. 

To learn more, visit the [Functions Copilot documentation](/docs/connections/functions/copilot/).

> #### IP Allowlisting for functions

Reviewer: ```suggestion
#### IP Allowlisting
```

> With IP Allowlisting, users can ensure that outbound traffic from Functions will originate only from a predefined set of IP addresses. This provides added security for external services by making it easier to recognize and allowlist trusted traffic sources.

Reviewer: ```suggestion

> warning " "
> IP Allowlisting is only available for customers on Business Tier plans.

IP Allowlisting uses a NAT gateway to route outbound Functions traffic from Segment’s servers to your destinations through a limited range of IP addresses, which can prevent malicious actors from establishing TCP and UDP connections with your integrations.
```

Author: The warning `IP Allowlisting is only available for customers on Business Tier plans` has been added as a simple text as it is being displayed as  simple text in Destinations IP Allowlisting doc as well.
Ref: https://segment.com/docs/connections/destinations/#ip-allowlisting

> To learn more, visit [IP Allowlisting documentation](/docs/connections/destinations/#ip-allowlisting).

Reviewer: Should we remove the note in the Destination Overview page that mentions Destination Functions are not supported?

```suggestion
To learn more, visit [Segment's IP Allowlisting documentation](/docs/connections/destinations/#ip-allowlisting).
```
